core/controllers/contributor_dashboard.py

```diff
@@ -935,6 +936,89 @@ def get(
         self.render_json(self.values)
 
 
+class ContributorCertificateHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """Returns contributor certificate."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'username': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'has_length_at_most',
+                    'max_value': constants.MAX_USERNAME_LENGTH
+                }]
+            }
+        },
+        'suggestion_type': {
+            'schema': {
+                'type': 'basestring',
+                'choices': feconf.SUGGESTION_TYPE_CHOICES
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'from_date': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_regex_matched',
+                        'regex_pattern': constants.DATE_REGEX
+                    }]
+                }
+            },
+            'to_date': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_regex_matched',
+                        'regex_pattern': constants.DATE_REGEX
+                    }]
+                }
+            },
+            'language': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_supported_audio_language_code'
+                    }]
+                },
+                'default_value': None
+            },
+        }
+    }
+
+    @acl_decorators.can_fetch_all_contributor_dashboard_stats
+    def get(
+        self,
+        username: str,
+        suggestion_type: str
+    ) -> None:
+        """Handles GET requests."""
+        language = None
+        assert self.normalized_request is not None
+        from_date = self.normalized_request['from_date']
+        to_date = self.normalized_request['to_date']
+
+        if suggestion_type == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT:
+            language = self.normalized_request['language']
```

Comment:
Why is this needed?

Reply:
If the `suggestion_type` is not `feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT`, there will not be a parameter with `language`.

Reply:
You can do
```suggestion
        language = self.normalized_request.get('language')
```